src/core/parser/PDFParser.ts

```diff
@@ -200,7 +210,13 @@ class PDFParser extends PDFObjectParser {
       this.bytes.next();
     }
 
-    if (failed) throw new PDFInvalidObjectParsingError(startPos);
+    if (failed) {
+      const errMesssage = `Failed to parse invalid object: ${ref}`;
+      this.context.parseWarnings.push(errMesssage);
+      console.warn(errMesssage);
+      if (this.throwOnInvalidObjectParsingError)
+        throw new PDFInvalidObjectParsingError(startPos);
+    }
 
     const end = this.bytes.offset() - Keywords.endobj.length;
```

Comment:
if it's failed, it means you should not remove the length of the endobj keyword.